x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -155,19 +156,23 @@ export const useFetchIndex = (
           .subscribe({
             next: (response) => {
               if (isCompleteResponse(response)) {
-                const stringifyIndices = response.indicesExist.sort().join();
-
-                previousIndexesName.current = response.indicesExist;
-                setLoading(false);
-                setState({
-                  browserFields: getBrowserFields(stringifyIndices, response.indexFields),
-                  docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
-                  indexes: response.indicesExist,
-                  indexExists: response.indicesExist.length > 0,
-                  indexPatterns: getIndexFields(stringifyIndices, response.indexFields),
+                Promise.resolve().then(() => {
+                  ReactDOM.unstable_batchedUpdates(() => {
```

Comment:
I thought React was batching updates automatically. is it not working here?

Reply:
@semd that's the part i'm a little unsure about. in react < 18, updates are only batched automatically in scopes "known to react" as described in this comment facebook/react/issues/16387#issuecomment-521623662 linked to in that article, and I think updates done in the subscribe.next of some of these container hooks might fall outside of "known scopes"

Reply:
If that's the case, I think that's also why some pages will randomly get stuck in a loading state, (haven't seen it happen in production builds of react though for some reason) but @opauloh and i would see setTimelineResponse work correctly here https://github.com/elastic/kibana/blob/main/x-pack/plugins/timelines/public/container/index.tsx#L226 consistently, and then the rest of the code that cleared the loading state here https://github.com/elastic/kibana/blob/main/x-pack/plugins/timelines/public/container/index.tsx#L239-#L241 just never seemed to run. cc: @spong i think this is the phantom hanging loading state we've discussed a few times btw

Reply:
Feels to me like there are too many `useState`s in `useTimelineEvents` that could be bundled into a `useReducer`. Wdyt? (As a follow-up PR)

Reply:
IMO we should do it in 1 big redux reducer and do all the data fetching in async middleware, and never worry about blocking the ui again! (at least with this request)

Reply:
Yes to all of that!

Reply:
I would also consider wrapping the whole thing in a `useQuery` (react-query)